zcash_primitives/src/transaction/components/transparent/builder.rs

```diff
@@ -40,6 +40,26 @@ impl fmt::Display for Error {
     }
 }
 
+pub trait TransparentInput {
+    fn outpoint(&self) -> &OutPoint;
+    fn coin(&self) -> &TxOut;
+}
+
+/// An uninhabited type that allows the type of [`Builder::inputs`]
+/// to resolve when the transparent-inputs feature is not turned on.
+#[cfg(not(feature = "transparent-inputs"))]
+pub enum InvalidTransparentInput {}
```

Comment:
Does this need to be public? AFAICT it is only used as a type inside this module, and otherwise is exposed as `impl TransparentInput` which does not require this type to be public (similarly to `TransparentInputInfo`).
```suggestion
enum InvalidTransparentInput {}
```